src/lamp_py/mssql/mssql_utils.py

```diff
@@ -0,0 +1,173 @@
+import os
+from typing import Any, Dict, List, Union
+
+import pyodbc
+import pandas
+import sqlalchemy as sa
+from sqlalchemy.orm import sessionmaker
+import pyarrow
+import pyarrow.parquet as pq
+
+from lamp_py.runtime_utils.process_logger import ProcessLogger
+
+
+def get_local_engine(echo: bool = False) -> sa.future.engine.Engine:
+    """
+    Get an SQL Alchemy engine that connects to a MSSQL RDS
+    """
+    process_logger = ProcessLogger("create_mssql_engine")
+    process_logger.log_start()
+    try:
+        # disable pyodbc pooling because sqlachemy does pooling
+        pyodbc.pooling = False
+
+        db_host = os.environ.get("TM_DB_HOST")
+        db_name = os.environ.get("TM_DB_NAME")
+        db_user = os.environ.get("TM_DB_USER")
+        db_password = os.environ.get("TM_DB_PASSWORD", None)
+        db_port = 1433
+
+        assert db_host is not None
+        assert db_name is not None
+        assert db_port is not None
+        assert db_user is not None
+        assert db_password is not None
+
+        process_logger.add_metadata(
+            host=db_host, database_name=db_name, user=db_user, port=db_port
+        )
+
+        connection = sa.URL.create(
+            "mssql+pyodbc",
+            username=db_user,
+            password=db_password,
+            host=db_host,
+            port=db_port,
+            database=db_name,
+            query={
+                "driver": "ODBC Driver 18 for SQL Server",
+                "TrustServerCertificate": "yes",
+            },
+        )
+
+        engine = sa.create_engine(
+            connection,
+            echo=echo,
+        )
+
+        process_logger.log_complete()
+        return engine
+    except Exception as exception:
+        process_logger.log_failure(exception)
+        raise exception
+
+
+class MSSQLManager:
+    """
+    manager class for rds application operations
+    """
```

Comment:
All of the methods in here worked for me locally

Reply:
these all worked for me as expected.